tensorbay/opendataset/CACD/loader.py

```diff
@@ -0,0 +1,116 @@
+from ...utility import NameList
+from .._utility import glob
+
+DATASET_NAME = "CACD"
+_ATTRIBUTES = {
+    "attributes": [
+        {"name": "age", "type": "number", "minimum": 14, "maximum": 62},
+        {"name": "birth", "type": "number", "minimum": 1951, "maximum": 1990},
+        {"name": "range", "type": "number", "minimum": 1, "maximum": 50},
+        {"name": "year", "type": "number", "minimum": 2004, "maximum": 2013},
+        {"name": "name", "type": "string"},
+        {
+            "name": "lfw",
+            "type": "boolean",
+            "description": "Whether the celebrity is in LFW dataset.",
+        },
+    ]
+}
+_MAT_KEYS = ("name", "identity", "age", "birth", "lfw", "rank", "year")
+
+
+def CACD(path: str) -> Dataset:
+    """Dataloader of `Cross-Age Celebrity Dataset (CACD)`_ dataset.
+
+    .. _Cross-Age Celebrity Dataset (CACD): https://bcsiriuschen.github.io/CARC/
+
+    The file structure should be like::
+
+        <path>
+            CACD2000/
+                14_Aaron_Johnson_0001.jpg
+                ...
+            celebrity2000.mat
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.catalog.classification = _get_subcatalog()
+    segment = dataset.create_segment()
+    image_files = glob(os.path.join(root_path, "CACD2000", "*.jpg"))
+    labels_map = _get_labels_map(os.path.join(root_path, "celebrity2000.mat"))
+    for image in image_files:
+        category, attribute = labels_map[os.path.basename(image)]
+        image_data = Data(image)
+        image_data.label.classification = Classification(category, attribute)
+        segment.append(image_data)
+    return dataset
+
+
+def _get_labels_map(path: str) -> Dict[str, Tuple[str, Dict[str, Any]]]:
+    """Get celebrity_image_data from .mat file.
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Raises:
+        ModuleImportError: When the module "h5py" can not be found.
+
+    Returns:
+        A Dict of attributes.
+
+    """
+    try:
+        from h5py import File  # pylint: disable=import-outside-toplevel
+    except ModuleNotFoundError as error:
+        raise ModuleImportError(error.name) from error  # type: ignore[arg-type]
+    mat_file = File(path, "r")
+    celebrity_image_data = mat_file["celebrityImageData"]
+    celebrity_data = mat_file["celebrityData"]
+
+    # Name is a h5r object which can be searched in .mat file.
+    id2name_map = {
+        identity: _hdf5_to_str(mat_file[name])
+        for identity, name in zip(celebrity_data["identity"][0], celebrity_data["name"][0])
+    }
+    labels_map = {}
+    # The "name" is not the name of the star but the name of the image file.
+    for name, identity, *attrs in zip(*(celebrity_image_data[key][0] for key in _MAT_KEYS)):
```

Comment:
in our SDK, `attr` means the python attr, not the attribute in label.
maybe rename it to `values`